react/src/App.js

```diff
@@ -20,35 +20,32 @@ class App extends Component {
       perfData: [],
       treeData: null,
       rows: null
-
     };
     // Binding methods for tree manipulation
     this.onClick = this.onClick.bind(this);
   }
   //click events for each node performance
 
   onClick(e) {
-    console.log('perData', this.state.perfData)
+    this.setState({ rows: null })
     const myDisplay = this.state.perfData.filter(value => {
-      console.log('value', value)
       return e.name == value.route;
     });
-    console.log(e.name)
-    // console.log("myDisplaysss", myDisplay);
-    //This is coping the data from
 
-    if (myDisplay.length > 0) {
-      console.log('row', rows)
-      rows = myDisplay.slice();
-      let method = rows[0].method;
-      this.setState({ rows });
+    if (myDisplay.length) {
+      this.setState({ rows: myDisplay });
     }
   }
   componentDidMount() {
     socket.on('tree', tree => {
       console.log('trees', tree)
       this.setState({ treeData: [tree.root] })
     })
+    socket.on('data', data => {
+      this.setState(prevState => {
+        prevState.perfData.push(data)
+      })
+    });
     if (this.state.treeData) {
       const dimensions = this.treeContainer.getBoundingClientRect();
```

Comment:
Why is this here? `this.state.treeData` will always be false because the data is not there when the component renders. If this is needed, it needs to go outside `componentDidMount`, possibly inside `render()` method.

Reply:
This was for positioning for the D3 Canvas, Currently not working. I am going to position the Tree starting point to middle of the Canvas rather than Left Corner.

Reply:
👍